fix(compiler-cli): don't emit empty providers array

crisbeto opened the pull request:
Saves us some bytes by not emitting `providers` in `defineInjector`. While the amount of bytes isn't huge, I think that this change is worthwhile, because `ng generate` currently generates `providers: []` with every `NgModule` which users can forget to remove.

AndrewKushnir reviewed (approved):
Looks great, thanks @crisbeto 👍

AndrewKushnir commented on packages/compiler-cli/src/ngtsc/annotations/ng_module/src/handler.ts:
```diff
@@ -332,11 +332,16 @@ export class NgModuleDecoratorHandler implements
+    // In most cases the providers will be an array literal. Check if it has any elements
+    // and don't include the providers if it doesn't which saves us a few bytes.
+    if (rawProviders !== null &&
+        (!ts.isArrayLiteralExpression(rawProviders) || rawProviders.elements.length > 0)) {
```
One more comment: @crisbeto do we have a test to check the `!ts.isArrayLiteralExpression(rawProviders)` condition? If not, could you please add one.

crisbeto replied:
Done.

crisbeto committed:
fix(compiler-cli): don't emit empty providers array

Saves us some bytes by not emitting `providers` in `defineInjector`. While the amount of bytes isn't huge, I think that this change is worthwhile, because `ng generate` currently generates `providers: []` with every `NgModule` which users can forget to remove.

jessicajaniuk commented:
This PR was merged into the repository by commit 04acc6b14d502597a98537019822f1a1217c9a61.
